specs/Picker-API_spec.md

```diff
@@ -0,0 +1,121 @@
+- Windows.Storage.Pickers.FolderPicker (https://docs.microsoft.com/en-us/uwp/api/windows.storage.pickers.folderpicker) 
+
+# Description
+
+The goal of this API is to address the gaps in the existing File/Folder picker APIs. The API surface is similar to the existing
+Picker APIs in Windows SDK with additional methods to support File + Folder picker and multiple folder picker. The Project Reunion APIs will supercede the Windows APIs.
+
+# Examples
+## Show file and folder picker
+```c#
+    var picker = new StorageItemPicker();
+    picker.FiltersAndDescriptions.Insert("Images", "*.jpg, *.png");
+    picker.StartLocation = startLocation;
+    picker.PickerKinds = StorageItemPickerKinds.Folder | StorageItemPickerKinds.FileOpen;
+    // multi-select will be automatically set
+    IReadOnlyList<Windows.Storage.IStorageItem> storageItems = await picker.ShowPickerAsync();
+
+    if(0 < storageItems.Count)
+    {
+    }
+```
+## Show single folder picker
+```c#
+    var picker = new StorageItemPicker();
+    picker.PickerKinds = StorageItemPickerKinds.Folder;
+    IReadOnlyList<Windows.Storage.IStorageItem> folder = await picker.ShowPickerAsync();
+```
+## Show multi folder picker
+```c#
+    var picker = new StorageItemPicker();
+    picker.PickerKinds = StorageItemPickerKinds.Folder;
+    picker.MultiSelect = true;
+    IReadOnlyList<Windows.Storage.IStorageItem> folders = await picker.ShowPickerAsync();
+```
+## Show multiple file picker
+```c#
+    var picker = new StorageItemPicker();
+    picker.StartLocation = startLocation;
+    picker.MultuSelect = true;
+    picker.PickerKinds = StorageItemPickerKinds.FileOpen;
+    IReadOnlyList<Windows.Storage.IStorageItem> files = await picker.ShowPickerAsync();
+```
+## Show File Save picker
+```c#
+    var picker = new StorageItemPicker();
+    picker.PickerKinds = StorageItemPickerKinds.FileSave;
+    IReadOnlyList<Windows.Storage.IStorageItem> fileToSave = await picker.ShowPickerAsync();
+```
+
+# Remarks
+# API Notes
+# API Details
+```c#
+    namespace Microsoft.Reunion.Storage.Pickers
+    {
+        enum PickerViewMode
+        {
+            List = 0,
+            Details,
+            Tiles,
+            Content
+        };
+
+        enum StorageItemPickerKinds
+        {
+            FileOpen = 0x1,
+            Folder = 0x2,
+            FileSave = 0x4
+        };
+
+        runtimeclass StorageItemPicker
+        {
+            StorageItemPicker();
+
+            /// Properties
+            String StartLocation;
+            String CommitButtonText;
+            String CancelButtonText;
+            Windows.Foundation.Collections.IPropertySet FiltersAndDescriptions{ get; };
+            ViewMode PickerViewMode;
+            StorageItemPickerKinds PickerKinds;
+            String SuggestedSaveFile;
+            String DefaultFileExtension;
+            Boolean MultiSelect;
+            Int64 Result{ get; }
```

Comment:
What is this?

Reply:
This holds the dialog result(Canceled, Access_Denied etc.). This can be removed in favor of packaging the result with IAsyncOperation

---
In reply to: [499730295](https://github.com/microsoft/ProjectReunion/pull/99#discussion_r499730295) [](ancestors = 499730295)